From cd63169cb94402537688e1e79293521abc65ae88 Mon Sep 17 00:00:00 2001
From: Rameshkumar Sundaram <quic_ramess@quicinc.com>
Date: Thu, 3 Feb 2022 21:25:29 +0530
Subject: [PATCH] ath11k: add pci3 and pci4 domain names to dp/ce_irq_name

Alder+Pine+Pine(IPQ9574) platform enumerates the Pine PCI cards
with domain id 3(Pine 6G) and 4(Pine 5G).
Increase MAX PCI domains to 5 and add corresponding
irq names to dp/ce_irq_name arrays.
This will be helpful to uniquely identify irqs
registered by each Pine PCI cards as below,

 pci3_wbm2host_tx_completions_ring1
 pci3_wbm2host_tx_completions_ring2
 pci3_wbm2host_tx_completions_ring3
 pci3_lmac_reo_misc_irq
 pci3_reo2host_destination_ring1
 ..
 pci4_wbm2host_tx_completions_ring1
 pci4_wbm2host_tx_completions_ring2
 pci4_wbm2host_tx_completions_ring3
 pci4_lmac_reo_misc_irq
 pci4_reo2host_destination_ring1
 ..

and therefore enable affinity scripts to work properly.

Signed-off-by: Rameshkumar Sundaram <quic_ramess@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/pci.h     |  2 +-
 drivers/net/wireless/ath/ath11k/pci_cmn.c | 18 ++++++++++++++++++
 drivers/net/wireless/ath/ath11k/pci_cmn.h |  1 -
 3 files changed, 19 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/pci.h
+++ b/drivers/net/wireless/ath/ath11k/pci.h
@@ -24,7 +24,7 @@
 
 /* register used for handshake mechanism to validate UMAC is awake */
 #define PCIE_SOC_WAKE_PCIE_LOCAL_REG		0x3004
-#define ATH11K_MAX_PCI_DOMAINS 2
+#define ATH11K_MAX_PCI_DOMAINS			5
 
 #define PCIE_PCIE_PARF_LTSSM			0x1e081b0
 #define PARM_LTSSM_VALUE			0x111
--- a/drivers/net/wireless/ath/ath11k/pcic.c
+++ b/drivers/net/wireless/ath/ath11k/pcic.c
@@ -16,6 +16,15 @@ const char *ce_irq_name[ATH11K_MAX_PCI_D
 		ATH11K_PCI_CE_IRQS_NAME(1)
 	},
 	{
+		ATH11K_PCI_CE_IRQS_NAME(2)
+	},
+	{
+		ATH11K_PCI_CE_IRQS_NAME(3)
+	},
+	{
+		ATH11K_PCI_CE_IRQS_NAME(4)
+	},
+	{
 		ATH11K_PCI_CE_IRQS_NAME()
 	},
 };
@@ -28,6 +37,15 @@ const char *dp_irq_name[ATH11K_MAX_PCI_D
 		ATH11K_PCI_DP_IRQS_NAME(1)
 	},
 	{
+		ATH11K_PCI_DP_IRQS_NAME(2)
+	},
+	{
+		ATH11K_PCI_DP_IRQS_NAME(3)
+	},
+	{
+		ATH11K_PCI_DP_IRQS_NAME(4)
+	},
+	{
 		ATH11K_PCI_DP_IRQS_NAME()
 	},
 };
